### PR TITLE
Simplify TwoPanelState usage with new routing

### PR DESCRIPTION
**Pull request by toddburnside**
As well as (IMO) simplifying the code, it also seems to have fixed the back button issues.

### CODE REVIEW

File: explore/src/main/scala/explore/tabs/TargetTabContents.scala [hunk 1]
```diff
@@ -535,6 +530,18 @@ object TargetTabContents {
     ScalaFnComponent
       .withHooks[Props]
       .useStateViewWithReuse(TwoPanelState.initial[TargetOrObsSet](SelectedPanel.Uninitialized))
+      .useEffectWithDepsBy((props, state) =>
+        (props.focusedObsSet, props.focusedTarget, state.zoom(selectedLens).value.reuseByValue)
+      ) { (_, _) => params =>
+        val (focusedObsSet, focusedTarget, selected) = params
+        Callback.log(s"Tab contents $focusedObsSet, $focusedTarget, ${selected.get}") >>
```
Comment (cquiroz) [inline explore/src/main/scala/explore/tabs/TargetTabContents.scala:537]: Do we want to keep this?

**Review by cquiroz — APPROVED** [review]
LGTM, great improvement

File: explore/src/main/scala/explore/tabs/TargetTabContents.scala (hunk 1 above)
Comment (toddburnside) [reply explore/src/main/scala/explore/tabs/TargetTabContents.scala:537]: Nope. Just some debugging I missed.

**Review by rpiaggio — APPROVED** [review]
Very nice. LGTM

**Pull request merged**